.github/actions/tp-integration-tests/entrypoint.sh

```
@@ -141,82 +144,99 @@ start_traffic_vault &
-./traffic_ops_golang --cfg ./cdn.conf --dbcfg ./database.conf -riakcfg riak.conf &
-tail -f warning.log 2>&1 | color_and_prefix "${yellow_bg}" 'Traffic Ops' &
-tail -f error.log 2>&1 | color_and_prefix "${red_bg}" 'Traffic Ops' &
-tail -f event.log 2>&1 | color_and_prefix "${gray_bg}" 'Traffic Ops' &
 
-cd "../../traffic_portal"
+cd "${REPO_DIR}/traffic_portal/test/integration"
 npm ci
-bower install
-grunt dist
-
-cp "${resources}/config.js" ./conf/
-touch tp.log access.log
-sudo forever --minUptime 5000 --spinSleepTime 2000 -l ./tp.log start server.js &
-
-fqdn="https://localhost:8443"
-while ! curl -Lvsk "${fqdn}/api/3.0/ping" >/dev/null 2>&1; do
-  echo "waiting for TP/TO server to start on '${fqdn}'"
-  sleep 10
-done
+PATH=$(pwd)/node_modules/.bin/:$PATH
 
+webdriver-manager update --gecko false --versions.chrome "LATEST_RELEASE_$CHROME_VER"
 
-cd "test/end_to_end"
 jq " .capabilities.chromeOptions.args = [
-    \"--disable-extensions\",
-    \"--disable-gpu\",
     \"--headless\",
     \"--no-sandbox\",
+    \"--disable-gpu\",
     \"--ignore-certificate-errors\"
-  ] | .baseUrl = \"${fqdn}\" | del(.seleniumAddress)" \
-  conf.json > conf.json.tmp && mv conf.json.tmp conf.json
-
-onFail() {
-	docker logs "$trafficvault" 2>&1 |
-		color_and_prefix "$gray_bg" 'Traffic Vault';
-  cat tp.log | color_and_prefix "${gray_bg}" 'Forever'
-  cat access.log | color_and_prefix "${gray_bg}" 'Traffic Portal'
-  exit 1
-}
-
-sudo protractor ./conf.js || onFail
+  ] | .params.apiUrl = \"${tp_fqdn}/api/4.0\" | .params.baseUrl =\"${tp_fqdn}\"
+  | .capabilities[\"goog:chromeOptions\"].w3c = false | .capabilities.chromeOptions.w3c = false" \
+  config.json > config.json.tmp && mv config.json.tmp config.json
+  
+tsc
+
+# Wait for tp/to build
+timeout 5m bash <<TMOUT
+  while ! curl -Lvsk "${tp_fqdn}/api/4.0/ping" >/dev/null 2>&1; do
+    echo "waiting for TP/TO server to start on '${tp_fqdn}'"
+    sleep 10
+  done 
+TMOUT
```

If this command times out, can we fail early instead of running the tests?

---

Timeout returns non-zero when the command times out.  As can be seen in this run:
https://github.com/shamrickus/trafficcontrol/runs/2168798570?check_suite_focus=true#step:5:1096